### DISCUSSION TITLE
Add weather reporting stations! and more suggestions that some dont make sense

### DISCUSSION THREAD

**Discussion by rice-with-beans** [niels747/2D-Weather-Sandbox#21] [Ideas]
So the graph which can be turned on by pressing 'g' is great, but it does not show the temperature everywhere and I want something new that you can add to the simulation and it stays there however it can be hidden by pressing keys of ur choice. I'll show you an example. It should also predict the weather like for example. If the dew point is low and the temperature is high, low chance of precipitation. But, if the dew point is high and temp is high, high chance of precipitation. heh. Also, would it be possible to speed up day/night cycles. I used a stopwatch and it took 40 minutes for a full day. imo that is a very long time and I want there to be a way to shorten it to like 5 minutes because I would manually change the sun angle and I would forget to change it upsetting the balance of day and night. Now, one more annoying suggestion to add to your stress, Can there be a tool to shift solar noon? In game the highest sun angle is at 12 pm, however in reality it is almost never 12 pm. For example in high latitudes in winter solar noon is around 10 - 11 am and solar noon in california in summer is like 1:30 pm. none of this makes sense
![image](https://user-images.githubusercontent.com/113717680/191576841-26423338-7fb9-4759-8cb0-c61a6f2249b0.png)

**Comment by niels747** [comment]
Thank you for your input. Weather stations are a good idea and I will add it to the todo list. I'm not sure about the prediction part though...
I will also add an option to change the day/night cycle speed. Right now a day takes a constant number of iterations/simulation steps: 240000
Each iteration the time is increased by 0.0001 here: https://github.com/niels747/2D-Weather-Sandbox/blob/369554f0b472dbee5d2cc8af5c23f702991c2034/app.js#L1945

That a day took 40 minutes for you is because your computer is very slow or you are running at a very high resolution.
Making the day go faster would mean there is less simulation steps per day and there would be less temperature difference bewteen day and night, less realistic. You might say just increase the rate of cooling and heating, but then the temperature would change to fast relative to the airflow and convection rate wich would be unrealistic.

Now about the shifting solar noon:
There are timezones on earth and unless you are exactly at the right longditude within them the sun is never gonna be exactly at the highest point at 12:00
For the simulation let's just pretent you are at the perfect place in the timezone as i don't see the use of another slider wich doesn't really influence the simulation itself(the ui is getting cluttered). Here you can see the timezones and the ofsets: https://i.stack.imgur.com/IaWgC.png
